src/raster/r.maxent.train/r.maxent.train.py

```diff
@@ -0,0 +1,1125 @@
+# % answer: 0.5
+# % options: 0-1
+# % guisection: Advanced
+# %end
+
+# %flag
+# % key: d
+# % label: Keep duplicate presence records.
+# % description: Keep duplicate presence records. If environmental data are in grids, duplicates are records in the same grid cell. Otherwise, duplicates are records with identical coordinates.
+# % guisection: Advanced
+# %end
+
+# %flag
+# % key: s
+# % label: Use a random seed
+# % description: If selected, a different random seed will be used for each run, so a different random test/train partition will be made and a different random subset of the background will be used, if applicable.
+# % guisection: Advanced
+# %end
+
+# %flag
+# % key: x
+# % label: Add all samples to the background
+# % description: Add all samples to the background, even if they have combinations of environmental values that are already present in the background
+# % guisection: Advanced
+# %end
+
+# %option G_OPT_F_BIN_INPUT
+# % key: maxent
+# % label: Location Maxent jar file
+# % description: Give the path to the Maxent executable file (maxent.jar)
+# % required: no
+# %end
+
+# %option G_OPT_M_NPROCS
+# % key: threads
+# % label: Number of processor threads to use.
+# %end
+
+# %option G_OPT_MEMORYMB
+# % Description: Maximum memory to be used by Maxent (in MB)
+# %end
+
+# %flag
+# % key: v
+# % label: Make the Maxent user interface visible
+# % description: Show the Maxent interface and do not start automatically. Use for debugging.
+# %end
+
+# %flag
+# % key: m
+# % label: Do not autorun Maxent
+# % description: When you select this option, Maxent will not start before you hit the start option. This will set the -v flag to true (overriding user choice).
+# %end
+
+# %flag
+# % key: i
+# % label: Copy maxent.jar to addon directory
+# % description: copy the maxent.jar (path provided with the 'maxent' parameter) to the addon scripts directory.
+# %end
+
+# %flag
+# % key: u
+# % label: Overwrites maxent.jar in addon directory
+# % description: copy the maxent.jar (path provided with the 'maxent' parameter) to the addon scripts directory. If the file already exist in the addon directory, it is overwritten.
+# %end
+
+# %rules
+# % exclusive: replicates,randomtestpoints
+# % exclusive: -i, -u
+# %end
+
+# import libraries
+# ------------------------------------------------------------------
+import atexit
+import csv
+import re
+import os
+import shutil
+import subprocess
+import sys
+import uuid
+import numpy as np
+import grass.script as gs
+
+
+CLEAN_LAY = []
+
+
+# Funtions
+# ------------------------------------------------------------------
+def find_index_case_insensitive(lst, target):
+    """
+    Find index for string match, matching case insensitive
+    """
+    for i, item in enumerate(lst):
+        if item.lower() == target.lower():
+            return i
+    return -1  # Return -1 if the element is not found
+
+
+def Check_if_layer_exist(layer):
```

Comment:
```suggestion
def check_if_layer_exist(layer):
```

Python functions normally don't start with an uppercase letter. The color of the syntax highlighting is different than for a function here.

The changes need to be done for all references too.

Reply:
Function was not used, so removed it.